### PR TITLE
feat(streamlit): run approved finite batches through the existing CLI

### PR DESCRIPTION
**Pull request by liq22**
## User-visible change
After validating a base experiment, preview a bounded parameter grid with exact YAMLs and fit counts, then submit it explicitly. The existing run service executes trials serially, pauses on the first failure, and supports explicit continuation of pending work or cancellation. Each trial retains its ordinary CLI outputs.

## Invariant
An approved finite plan executes through the same inspector/preflight/CLI and cancellation path as a single run, without parameter repair, silent retries, duplicate submission on page reruns, or interleaving another run. An explicitly selected historical trial is opened by its exact run ID, never replaced with a recent run.

## Scope
Eleven changed paths: optional frontend/services, existing UI workflow test selection, focused/integration/AppTest coverage, README and doc/changelog/2026-09-09-streamlit-serial-batches.md. Temporary transfer files are absent from the final diff. No backend configuration, Factory, Pipeline, model, split, estimator, checkpoint-selection or experiment-YAML changes. Research PR #231 remains separate.

## Behavior and limits
- First UI varies learning rate, batch size, epochs, seed and iterations; max 16 CLI calls / 64 fits, with lower user limits.
- All snapshots pass public inspection before any trial; public preflight repeats at each launch on the exact YAML.
- One worker reserves its single-run slot throughout the batch, including paused work.
- Continue submits pending trials only; failed trials remain failed. Cancel stops the managed child and cancels pending items.
- Page reruns only monitor; server restart marks the batch interrupted without resubmission.
- batch.json stores scheduling facts and approved configs, not a second scientific result or evaluator.
- No parallel/multi-worker scheduling, crash recovery, adaptive test-guided search, Agent callbacks, external-data downloads or benchmark promotion.

## Validation
Earlier local execution: lightweight service tests 111 passed; isolated real two-trial CPU Dummy batch passed, distinct checkpoint/metrics/summary roots and exact execution YAMLs; docs validation and diff checks passed. The historical-trial correction added two before/after regression tests.
Final head 9d08ea6dc08c9e41d46fbc56a900ae6d86adac8f has all four PR workflows completed successfully: Streamlit 34333363117; Core 34333363174; Layout 34333363187; Submodule 34333363121. Streamlit includes real public-inspector/Dummy and page AppTest execution; these are CI results, not claimed local browser tests. The historical-trial review thread is resolved after inspecting the exact tests and final-head checks.

## Remaining work
Detached process reconciliation and editor-independent monitoring remain separate C1 frontend fixes. Process completion is not scientific validation.

Rollback: revert the squash commit.

### CODE REVIEW

**Review by liq22 — COMMENTED** [review]
Bounded review update, not an independent reviewer count:

- Scope: the final diff contains only 9 frontend/test/documentation paths. No training backend or experiment YAML changed; temporary transfer files are absent.
- Local final lightweight suite: 111 passed. The complete real public-contract file also passed locally: 4 passed in 57.12s, including an ordinary run followed by the two-trial batch.
- First CI candidate: Linux/Windows focused suites and all four real public/Dummy integration tests passed. AppTest found a genuine refresh-state defect after clearing a submitted preview (4 page tests passed, 1 failed).
- The current patch keeps the inspected plan visible and marks it submitted. Run remains disabled until an explicit new Preview; the regression still asserts that a further page rerun cannot resubmit. No test was skipped and the runtime contract was not weakened.
- Serial execution uses existing public analysis/preflight/start/get/cancel, reserves one process-local slot across trials and while paused, and retains failure history when the user chooses to continue pending items.
- Scope limits remain: single worker, no crash resume, no auto retry, no adaptive tuning and no benchmark-validity promotion. Browser disconnect behavior follows worker lifetime; it has not been tested with a real disconnected browser.

Current-head CI must pass before merge.

File: apps/streamlit/ui_batch.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    if right.button("Cancel batch", key=f"cancel-batch::{record.batch_id}",
+                    disabled=record.is_terminal):
+        cancel_batch(repo_root, record.batch_id)
+        st.rerun()
+    st.caption("Continuing runs only pending trials. Failed trials remain failed and are not retried.")
+    for trial in record.trials:
+        if trial["run_id"] and st.button(f"View trial {trial['index']}",
+                                        key=f"batch-run::{record.batch_id}::{trial['index']}"):
+            st.session_state.selected_run_id = trial["run_id"]
+            st.session_state.active_run_id = trial["run_id"]
```
[the remark this answers is not in the document]
Comment (liq22) [reply apps/streamlit/ui_batch.py:49]: Confirmed with two regressions: an old trial outside the recent 20 opened the newest run, and a missing explicitly selected trial was silently replaced. Both fail against the previous selector. Fixed in 9d08ea6: load the explicitly requested ID with get_run and include it in the selector; when unavailable, retain the selected ID and display its error instead of another experiment. No scan-limit increase or second result authority. Local refresh-policy tests: 6 passed (including the two new regressions); previous implementation: 2 expected assertion failures. Current-head CI is running.
